packages/augur-test/src/libs/MockMeshServer.ts

```diff
@@ -87,22 +88,41 @@ export class MockMeshServer {
                 let response = "";
                 if (jsonRpcRequest.method == "mesh_getOrders") response = self.getOrders(jsonRpcRequest.id, jsonRpcRequest.params);
                 else if (jsonRpcRequest.method == "mesh_addOrders") response = self.addOrders(jsonRpcRequest.id, jsonRpcRequest.params, connection);
-                else if (jsonRpcRequest.method == "mesh_subscribe") response = self.subscribe(jsonRpcRequest.id);
+                else if (jsonRpcRequest.method == "mesh_subscribe") response = self.subscribe(jsonRpcRequest.id, jsonRpcRequest.params, connection);
                 else throw new Error(`Bad Request: ${jsonRpcRequest.method}`);
                 connection.sendUTF(response);
             }) as any);
         }) as any);
     }
 
-    subscribe(id: number): string {
+    subscribe(id: number, params: string[], connection: WebSocket.connection): string {
+        let subscription =  "0xdeadbeefdeadbeefdeadbeefdeadbeef";
+        const heartbeatSub =  "0xab1a3e8af590364c09d0fa6a12103ada";
+
+        if (params[0] === "heartbeat") {
+            subscription = heartbeatSub;
+            this._subInterval = setInterval(() => {
```

Comment:
If we mess up and subscribe twice then this will create a second interval function that can't be cleared.

Reply:
Good call -- I can error if it's being called twice.